### PR TITLE
Helm, remove unused config and add if statements to be able to disable a few resources

### PR DESCRIPTION
**Pull request by NissesSenap**
There a number of clusterroles that isn't needed for this controller so I removed them.
I couldn't find any kubebuilder definition of the rbac rules so I assume they are manually created.

I also added if statments so you can disable PSP since it will be deprecated in 1.25 and there are already many other options to PSP.
You can also disable networkpolicy now, It should definitely be on by default by sadly there are CNI:s that don't support networkpolicys.

Removed the configmap that wasn't used and gave an option to disable the usage for pull secret.

## Checklist

- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by NissesSenap** [comment]
First of all thanks for a great project, I have looked at the generated metrics and it looks like a good use case for us.

This is my first time contributing to a giantswarm repo so I don't know if you want all the commits to be squashed in to one or not. 
This PR could definitely be two seperate ones, I hope you think it's okay like this. If not I can of course create two PR:s.

**Comment by stone-z** [comment]
@NissesSenap when you have a chance, try pushing to your fork again to see if it will re-prompt me to run CI. Once CI runs I can test it on our side. Worst case I'll have you open a new PR

**Comment by NissesSenap** [comment]
I can at least see the approval info, so lets hope you can as well.

image

After fixing the merge issue I did some more cleanup and also added some small instructions on how to use the helm chart.

File: README.md [hunk 1]
```diff
@@ -81,3 +81,13 @@ exporter:
...
 [starboard-upstream]: https://github.com/aquasecurity/starboard
+
+## Helm
+
+How to install the starboard-exporter using helm:
+
+```shell
+helm repo add giantswarm https://giantswarm.github.io/giantswarm-catalog
+helm repo update
+helm upgrade -i starboard-exporter --namespace starboard-operator giantswarm/starboard-exporter
```
Comment (stone-z) [inline README.md:92]:
```suggestion
helm upgrade -i starboard-exporter --namespace <starboard namespace> giantswarm/starboard-exporter
```

File: helm/starboard-exporter/values.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@ registry:
 image:
   name: "giantswarm/starboard-exporter"
   tag: "[[ .Version ]]"
+  pull_secret_enabled: true
```
Comment (stone-z) [inline helm/starboard-exporter/values.yaml:6]:
```suggestion
```
I think the user should have to opt-in here, not give them a blank secret by default

**Review by stone-z — COMMENTED** [review]
Ok I changed some more settings so hopefully after these updates CircleCI will let us run now

File: helm/starboard-exporter/values.yaml (hunk 2 above)
Comment (NissesSenap) [reply helm/starboard-exporter/values.yaml:6]: I personally don't have any need for it. I don't know if you do ether, it was just a default config that I assumed you wanted in all your helm charts. If you don't see any need I will remove it all together.

Comment (NissesSenap) [reply helm/starboard-exporter/values.yaml]: I pushed a commit where it's removed all together

**Review by stone-z — APPROVED** [review]
CircleCI is killing me. I tested this manually, LGTM. Thanks @NissesSenap :pray:

**Pull request merged**